xarray/core/dataset.py

```diff
@@ -1068,14 +1068,14 @@ def persist(self, **kwargs) -> "Dataset":
     @classmethod
     def _construct_direct(
         cls,
-        variables,
-        coord_names,
-        dims=None,
-        attrs=None,
-        indexes=None,
-        encoding=None,
-        close=None,
-    ):
+        variables: Mapping[Hashable, Variable],
+        coord_names: Set[Hashable],
+        dims: Mapping[Hashable, int] = None,
+        attrs: Mapping = None,
+        indexes: Mapping[Hashable, Index] = None,
+        encoding: Mapping = None,
```

Comment:
I think these can safely be `Dict`s, same as the type hints for `._replace`, because we aren't planning to swap out any of the `dims` etc. for a `CopyableMutableMapping`, only the `variables`.

Reply:
Is there a disadvantage to having wider types for the inputs though?

Reply:
I'll merge this for the moment but very happy to make the change later!